bundles/org.openhab.binding.casokitchen/src/main/java/org/openhab/binding/casokitchen/internal/handler/TwoZonesWinecoolerHandler.java

```diff
@@ -0,0 +1,255 @@
+                                updateState(new ChannelUID(thing.getUID(), GENERIC, LAST_UPDATE),
+                                        new DateTimeType(timestamp.atZone(timeZoneProvider.getTimeZone())));
+                                break;
+                            case HINT:
+                                updateState(new ChannelUID(thing.getUID(), GENERIC, HINT),
+                                        StringType.valueOf(result.hint));
+                                break;
+                        }
+                        break;
+                    case TOP:
+                    case BOTTOM:
+                        switch (channel) {
+                            case LIGHT:
+                                updateState(new ChannelUID(thing.getUID(), group, LIGHT),
+                                        OnOffType.from(result.light2));
+                                break;
+                            case POWER:
+                                updateState(new ChannelUID(thing.getUID(), group, POWER),
+                                        OnOffType.from(result.power2));
+                                break;
+                            case TEMPERATURE:
+                                updateState(new ChannelUID(thing.getUID(), group, TEMPERATURE),
+                                        QuantityType.valueOf(result.temperature2, SIUnits.CELSIUS));
+                                break;
+                            case TARGET_TEMPERATURE:
+                                updateState(new ChannelUID(thing.getUID(), group, TARGET_TEMPERATURE),
+                                        QuantityType.valueOf(result.targetTemperature2, SIUnits.CELSIUS));
+                                break;
+                        }
+                        break;
+                }
+            });
+        } else if (LIGHT.equals(channelUID.getIdWithoutGroup())) {
+            LightRequest lr = new LightRequest();
+            lr.technicalDeviceId = configuration.deviceId;
+            if (command instanceof OnOffType) {
+                lr.lightOn = OnOffType.ON.equals(command);
+                switch (group) {
+                    case GENERIC:
+                        lr.zone = 0;
+                        break;
+                    case TOP:
+                        lr.zone = 1;
+                        break;
+                    case BOTTOM:
+                        lr.zone = 2;
+                        break;
+                }
+                Request req = httpClient.POST(LIGHT_URL);
+                req.header(HttpHeader.CONTENT_TYPE, "application/json");
+                req.header(HTTP_HEADER_API_KEY, configuration.apiKey);
+                req.content(new StringContentProvider(GSON.toJson(lr)));
+                try {
+                    ContentResponse response = req.timeout(60, TimeUnit.SECONDS).send();
+                    int responseStatus = response.getStatus();
```

Comment:
Would be nice if you can extract the http request/response handling outside of this method. and make it somewhat generic for future use.

Reply:
post function extracted